### PR TITLE
CORE-9275: modified the job status listener so that it attempts to re…

### PR DESCRIPTION
**Pull request by slr71**
…establish the AMQP connection if if fails to record a job status update

There are better ways to detect broken AMQP connections, but they would require more refactoring than I have time for at the moment. For now, `job-status-listener` will attempt to record the job status update once. If the status update can't be recorded then `job-status-listener` will attempt to reestablish the connection and publish the job status update again. If the connection can't be reestablished or the status update can't be published after the second attempt then the job status listener will abort. We'll rely on Kubernetes to restart the service after it aborts.

### CODE REVIEW

**Review by johnworth — APPROVED** [review]
:shipit:

**Comment by slr71** [comment]
Thanks for the review!

File: main.go [hunk 1]
```diff
@@ -26,31 +26,42 @@ var log = logrus.WithFields(logrus.Fields{
 var (
 	cfgPath = flag.String("config", "", "Path to the configuration file.")
 	cfg     *viper.Viper
-
-	client *messaging.Client
 )
 
-// JobUpdatePublisher is the interface for types that need to publish a job
-// update.
-type JobUpdatePublisher interface {
```
Comment (ianmcorvidae) [inline main.go:35]: This interface was removed but is still used in function signatures

File: main.go [hunk 2]
```diff
@@ -144,16 +159,14 @@ func main() {
 
 	uri := cfg.GetString("amqp.uri")
 	exchange := cfg.GetString("amqp.exchange.name")
-	var err error
-	client, err = messaging.NewClient(uri, true)
+
+	publisher, err := NewDefaultJobUpdatePublisher(uri, exchange)
```
Comment (ianmcorvidae) [inline main.go:163]: This `NewDefaultJobUpdatePublisher` isn't defined anywhere.

Comment (slr71) [reply main.go:163]: D'oh! I forgot to check in the new source file.

**Pull request merged**